src/controllers/trainer.js

```diff
@@ -73,4 +73,27 @@ const updateTrainer = (req, res) => {
     .catch((error) => res.status(500).json(error));
 };
 
-module.exports = { getAllTrainers, getTrainerById, updateTrainer };
+const deleteTrainer = (req, res) => {
+  const { id } = req.params;
+
+  trainers.findByIdAndDelete(id)
+    .then((result) => {
+      if (!result) {
+        return res.status(404).json({
+          message: `Trainer with the id: ${id} was not found, please try with another one`,
+          error: true,
+        });
+      }
+      return res.status(200).json({
+        message: `Trainer with the id: ${id} was successfully deleted.`,
+      });
+    })
+    .catch((error) => res.status(400).json({
+      message: 'There was an mistake!',
+      error,
+    }));
+};
+
+module.exports = {
+  getAllTrainers, getTrainerById, updateTrainer, deleteTrainer,
```

Comment:
this generated a problem for me (it does not mean that it generated it for you).
to solve it the recommended identation is 
module.exports = {
 getAllTrainers, 
getTrainerById, 
updateTrainer, 
deleteTrainer,
};